frontend/src/App.tsx

```diff
@@ -1010,27 +1051,19 @@ export class App extends PureComponent<Props, State> {
       //
       // below, but that doesn't work because basePath may contain unescaped
       // regex special-characters. This is why we're stuck with the
-      // weird-looking double `replace()`.
-      pageName = document.location.pathname
-        .replace(`/${basePath}`, "")
-        .replace(new RegExp("^/?"), "")
-        .replace(new RegExp("/$"), "")
-    } else {
-      const { appHash, scriptRunId, scriptName } = this.state
-      this.clearAppState(appHash as string, scriptRunId, scriptName)
-
-      const qs = queryString ? `?${queryString}` : ""
-      const basePathPrefix = basePath ? `/${basePath}` : ""
-
-      const pageUrl = `${basePathPrefix}/${pageName}${qs}`
-      window.history.pushState({}, "", pageUrl)
+      // weird-looking triple `replace()`.
```

Comment:
Personally, I don't think this comment is super necessary.

Reply:
It's completely non-obvious to me why you'd do this instead of writing the more compact 

```javascript
.replace(new RegExp(`^/${basePath}/?`), "")
```

Did you just miss the initial part of the comment due to the diff being formatted weirdly?